### PR TITLE
Adding index python panel poc

### PR DESCRIPTION
**Pull request by minhtuev**
Index Python Panel POC v1

image

### CODE REVIEW

File: plugins/index-panel/__init__.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    return sorted(paths)
+
+
+class IndexPanel(foo.Panel):
+    @property
+    def config(self):
+        return foo.PanelConfig(
+            name="index_panel",
+            label="Index Management Panel",
+            description="A simple panel allows users to create and drop"
```
Comment (imanjra) [inline plugins/index-panel/__init__.py:124]: Should there be a `+` joining these two string?

File: plugins/index-panel/__init__.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+            " indexes for the dataset.",
+            help_markdown="My plugin",
```
Comment (imanjra) [inline plugins/index-panel/__init__.py:126]: ✂️

File: plugins/index-panel/__init__.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+                    (ctx.dataset.group_field + ".name", 1),
+                    (field_name, 1),
+                ]
+                self._create_index(ctx, index_spec, unique=unique)
+
+            ctx.ops.notify(f"Successfully created index {field_name}")
+            self.refresh(ctx)
+
+    def render(self, ctx):
+        conn = fod.get_db_conn()
```
Comment (imanjra) [inline plugins/index-panel/__init__.py:231]: Can we move db interactions to `on_load` or some event

File: plugins/index-panel/fiftyone.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: "@voxel51/index-panel"
+description: A panel for managing indexes in FiftyOne
+version: 0.0.15
+fiftyone:
+  version: "*"
+license: Apache 2.0
```
Comment (imanjra) [inline plugins/index-panel/fiftyone.yml:6]: ✂️

File: plugins/index-panel/__init__.py [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+        )
+
+        drop_index_menu.btn("drop_btn", label="Drop index", on_click=self.drop_index, variant="contained")
+        panel.btn(
+            "refresh_btn",
+            label="Refresh",
+            on_click=self.on_refresh_button_click,
+        )
+
+        return types.Property(panel, view=types.ObjectView())
```
Comment (imanjra) [inline plugins/index-panel/__init__.py:301]: `, view=types.ObjectView()` can be omitted

File: plugins/index-panel/__init__.py [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+            view=create_index,
+            label="Create Index Menu"
+        )
+        create_index_menu.btn("add_btn", label="Create index", on_click=self.create_index, variant="contained")
+
+        drop_index_menu = panel.menu(
+            "drop_menu", variant="square", color="secondary"
+        )
+        drop_index = types.DropdownView()
+        self.build_drop_index_view(ctx, drop_index)
```
Comment (imanjra) [inline plugins/index-panel/__init__.py:286]: Nice!

**Review by imanjra — APPROVED** [review]
Left some minor which can be address in a follow-up. Tested locally. LGTM 🚀

File: plugins/index-panel/fiftyone.yml (hunk 4 above)
Comment (minhtuev) [reply plugins/index-panel/fiftyone.yml:6]: Ah lol

**Pull request merged**